.github/workflows/pr-differences-mutants.yml

```diff
@@ -0,0 +1,363 @@
+name: PR Differences Mutants
+
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - ready_for_review
+    paths:
+      - "**.rs"
+
+concurrency:
+  group: pr-differences-${{ github.head_ref || github.ref || github.run_id }}
+  # Always cancel duplicate jobs
+  cancel-in-progress: true
+
+jobs:
+  # Check and output whether to run big (`stacks-node`/`stackslib`) or small (others) packages with or without shards
+  check-big-packages-and-shards:
+    name: Check Packages and Shards
+
+    runs-on: ubuntu-latest
+
+    outputs:
+      run_big_packages: ${{ steps.check_packages_and_shards.outputs.run_big_packages }}
+      big_packages_with_shards: ${{ steps.check_packages_and_shards.outputs.big_packages_with_shards }}
+      run_small_packages: ${{ steps.check_packages_and_shards.outputs.run_small_packages }}
+      small_packages_with_shards: ${{ steps.check_packages_and_shards.outputs.small_packages_with_shards }}
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - run: cargo install --version 23.12.2 cargo-mutants
+
+      - name: Relative diff
+        run: |
+          git diff origin/${{ github.base_ref }}.. > git.diff
+
+      - name: Remove deleted file's lines from git.diff file
```

Comment:
For the names of the steps, i'd recommend keeping them short and generic since the UI is harder to read with longer names. 
ex: 
```
- name: Update git diff
```

Reply:
the same comment applies to other steps - KISS